components/places/src/db/tx/mod.rs

```diff
@@ -0,0 +1,99 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+mod coop_transaction;
+
+use crate::api::places_api::ConnectionType;
+use crate::error::*;
+use coop_transaction::TimeChunkedTransaction;
+use rusqlite::Connection;
+use sql_support::{ConnExt, UncheckedTransaction};
+
+/// High level transaction type which "does the right thing" for you.
+/// Construct one with `PlacesDb::begin_transaction()`.
+pub enum PlacesTransaction<'conn> {
+    Sync(TimeChunkedTransaction<'conn>),
+    Write(UncheckedTransaction<'conn>),
+    Read(UncheckedTransaction<'conn>),
```

Comment:
Hmm, can we not expose this at all, so that trying to start a `PlacesTransaction` on a read-only connection immediately returns an error?

Reply:
From slack: Basically, my fear is that we'll ship with a crash from that because we didn't have great test coverage or something, when really taking a transaction out on a read conn is ultimately harmless (you wouldn't be able to mutate anything anyway, and would get an error if you tried).

How would you feel about a warning instead?